src/components/Redirect.jsx

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React from 'react';
+import withSideEffect from 'react-side-effect';
+import RouterRedirect from 'react-router-dom/Redirect';
+
+const testForExternal = (to: RouterTo): boolean => {
+	if (to instanceof URL) {
+		return true;
+	}
+	if (typeof to === 'string') {
+		return to.startsWith('http');
+	}
+	return false; // this is a React Router 'location'
+};
+
+type RouterTo = string | LocationShape | URL;
+type RedirectProps = {
+	to: RouterTo,
+	push?: boolean,
+};
+
+/*
+ * A routing component that, when rendered, will redirect the application to
+ * another internal route _or_ external URL. On the server, the redirect will
+ * be handled with a 302 redirect HTTP response to the browser. On the client,
```

Comment:
I started adding in that functionality this morning but it requires some non-trivial refactoring that I think would be easier to implement and review in a separate ticket - I'll do it today, but I'll make a new PR to keep this one simple and make it easier to see the code specifically relevant to 301/302 support